### DISCUSSION TITLE
How to use client-side context?

### DISCUSSION THREAD

**Discussion by gustavotoyota** [trpc/trpc#4181] [Q&A] [answered]
I'm trying to pass custom cookies in a specific request.
I see I can add some context using procedure options:

```js
const result = await trpcClient.sessions.refresh.mutate({ refresh: true }, {
  context: {
    cookie: `refreshToken=${refreshToken}; loggedIn=true`,
  },
});
```

How do I use this context?
I tried grabbing it from `fetch`, but it doesn't seem to show up there:
```js
fetch(url, options) {
  console.log('fetch', { url, options }); // No context inside options

  return fetch(url, {
    ...options,

    credentials: 'include',
  });
},
```
I'm not using Next by the way.

**Comment by KATT** [comment, marked as answer]
Got added today!

https://trpc.io/docs/links/httpLink#httplink-options

**Reply by KATT** [reply]
See #4175

**Reply by gustavotoyota** [reply]
Cool! Thank you!

**Reply by gustavotoyota** [reply]
I think I would need be able to access `op` from inside `fetch` as well to pass cookies to the browser in SSR:
```js
async fetch(url, options) {
  const response = await fetch(url, {
    ...options,

    credentials: 'include',
  });

  // Set Quasar SSR cookies

  // options.op.context.ssrContext.res.setHeader( // Something like this
  //   'set-cookie',
  //   response.headers['set-cookie']!,
  // );

  return response;
},
```

**Reply by gustavotoyota** [reply]
Or maybe it's possible in another way?

**Reply by gustavotoyota** [reply]
I made an issue for this problem: https://github.com/trpc/trpc/issues/4184
